frontend/src/components/activePage/ActiveUtils.tsx

```diff
@@ -0,0 +1,265 @@
+      notification.warning({
+        message: instance.metadata?.name,
+        description: `${instance.metadata?.name} deleted`,
+      });
+    }
+  } else {
+    const { tenantNamespace: tnm } = instance.metadata as any;
+    const { templateName } = instance.spec?.templateCrownlabsPolitoItTemplateRef
+      ?.templateWrapper?.itPolitoCrownlabsV1alpha2Template?.spec as any;
+    switch (status) {
+      case 'VmiOff':
+        if (
+          !instance.spec?.running &&
+          (tnm === tenantNamespace || role === WorkspaceRole.user)
+        ) {
+          notification.warning({
+            message: templateName,
+            description: (
+              <>
+                <div>
+                  Instance Name: <i>{instance.metadata?.name}</i>
+                </div>
+                <div>
+                  Status: <i>Stopped</i>
+                </div>
+              </>
+            ),
+          });
+        }
+        break;
+      case 'VmiReady':
+        if (
+          instance.status?.url &&
+          instance.spec?.running &&
+          (tnm === tenantNamespace || role === WorkspaceRole.user)
+        ) {
+          notification.success({
+            message: <strong>{templateName}</strong>,
+            description: (
+              <>
+                <div>
+                  Instance Name: <i>{instance.metadata?.name}</i>
+                </div>
+                <div>
+                  Status: <i>Started</i>
+                </div>
+              </>
+            ),
+            btn: instance.status?.url && (
+              <Button
+                type="success"
+                size="small"
+                onClick={() => window.open(instance.status?.url!, '_blank')}
+              >
+                Connect
+              </Button>
+            ),
+          });
+        }
+        break;
+    }
+  }
+};
+
+const filterId = (instance: Instance, search: string) => {
+  if (
+    (instance.tenantId && instance.tenantId.includes(search)) ||
+    search === ''
+  ) {
+    return true;
+  }
+  return false;
+};
+
+enum DropDownAction {
+  start = 'start',
+  stop = 'stop',
+  destroy = 'destroy',
+  connect = 'connect',
+  ssh = 'ssh',
+  upload = 'upload',
+  destroy_all = 'destroy_all',
+}
+
+const setInstanceRunning = async (
+  running: boolean,
+  instance: Instance,
+  instanceMutation: (
+    options?:
+      | MutationFunctionOptions<
+          ApplyInstanceMutation,
+          Exact<{
+            instanceId: string;
+            tenantNamespace: string;
+            patchJson: string;
+          }>
+        >
+      | undefined
+  ) => Promise<
+    FetchResult<ApplyInstanceMutation, Record<string, any>, Record<string, any>>
+  >
+) => {
+  try {
+    const result = await instanceMutation({
+      variables: {
+        instanceId: instance.name,
+        tenantNamespace: instance.tenantNamespace!,
+        patchJson: getInstancePatchJson({ running: !running }),
```

Comment:
```suggestion
        patchJson: getInstancePatchJson({ running }),
```
I would suggest this mutation _ensured_ the running state passed to it. The idea is to pass the desired state from above, not the current state of the instance.
Consider adding a JSDoc comment for this.